### DISCUSSION TITLE
Can't make DataPointTooltip work with the samples

### DISCUSSION THREAD

**Discussion by sergiofdezperea** [GSI-CS-CO/chart-fx#638] [Q&A] [answered]
Hello,

I'm trying to migrate an application from java swing to javafx. This library looks perfect to replace jfreechart.

I was playing with the plugins and i can't make work DataPointTooltip using SimpleChartSample, when i check the render getAxes it returns an empty array.

In FXSampler it doesn’t work either for me.

![Recording 2023-11-13 at 08 55 32](https://github.com/fair-acc/chart-fx/assets/18255523/c8ed5056-cec0-4be1-859b-682da5dc5836)

Can you give me advise?

**Comment by wirew0rm** [comment, marked as answer]
Hey, thanks for reporting, this seems to have been broken on the 11.3 major release, where the semantic on the `renderer.getAxes()` list changed slightly from always returning the used axes to only holding the axes explicitly set by the user.

As a quick workaround you can explicitly set the renderer axes to the ones defined in the chart:
``` java
renderer.getAxes().addAll(chart.getAxes()); // explicitly set renderer axes until Tooltip Plugin is fixed
```

This should work for all simple charts that only use a single renderer and get you started while we'll try to figure out how this should best be fixed. Ultimately the plugin should use the same logic that is used by the renderer, but currently these methods are at least partially defined in the individual implementations and not exposed through the interface, so we will probably have to pull them up.

**Reply by sergiofdezperea** [reply]
Thanks for the explanation and workaround.

**Reply by yezhengli-Mr9** [reply]
~Curious about the effect (any demo) for `new DataPointTooltip(...)` described in #431?
Dot not feel any `findNearestDataPointWithinPickingDistance(...)` provides substantial nearest `DataPoint` (version `11.2.5`, `11.3.0`, etc.) #639~
https://github.com/fair-acc/chart-fx/blob/cd202c7c9a0b40f1e66b6d0fe8fd93fffbec5435/chartfx-chart/src/main/java/de/gsi/chart/plugins/DataPointTooltip.java#L102-L115
https://github.com/fair-acc/chart-fx/blob/ed86318c5a1bde23d285633a94860648aee5fb32/chartfx-chart/src/main/java/io/fair_acc/chartfx/plugins/DataPointTooltip.java#L103-L116
Can see hover effect then.
